eBCSgen/Parsing/ParseBCSL.py

```diff
@@ -188,7 +189,7 @@ def to_side(self):
 REGULATIONS_GRAMMAR = """
     regulation_def: "type" ( regular | programmed | ordered | concurrent_free | conditional ) 
 
-    !regular: "regular" _NL+ (DIGIT|LETTER| "+" | "*" | "(" | ")" | "[" | "]" | "_" | "|" | "&")+ _NL*
+    !regular: "regular" _NL+ (DIGIT|LETTER| "+" | "*" | "(" | ")" | "[" | "]" | "_" | "|" | "&" | ";")+ _NL*
```

Comment:
This approach works, but what I had in mind was changing the grammar for regular expression so it actually allows you to write only regular expressions. Now you try to compile whatever is given and if it isn't RE, you raise an exception. This way the user has no information about what is the actual issue with their input expression. Or maybe I am wrong?

Reply:
The user receives information about the issue in the expression here: `raise RegulationParsingError(f"Invalid regular expression: {re}. Error: {e}")`.But that is probably not sufficient as the position of the mistake is mentioned with relevance only to the regex and not the model. 

If I understand it correctly I should edit or maybe even create a new part/variable for the regex grammar. And I would probably keep the `try` and `catch` with `regex.compile()`, so it would be bulletproof in case the grammar does not cover everything.

Reply:
Can you provide an example of an error when, for example, a closing right bracket is missing?

Reply:
For the input: 
```#! regulation
type regular
(r1_S;r1_T;r2|r1_T;r1_S;r2
```
Same as the original test but witthout the closing right bracket.

The parsing output was: `'Error trying to process rule "regular":\n\nError while parsing the regulation:\n\nInvalid regular expression: (r1_S;r1_T;r2|r1_T;r1_S;r2. Error: missing ) at position 26'`

Reply:
The thing is, there obviously is information about the particular parsing problem, but it won't properly propagate to the top-level error provided to the user (or the editor). That contains **particular position** of the error in the file, allowing us to pinpoint its location properly.

```
{
    "unexpected": [], # empty in this case, not generally I guess
    "expected": [")"],
    "line": XY,
    "column": 26,
},
```

I see two options for how to achieve this:

1. Modify the grammar so it "replicates" grammar used by `regex` package. The error would then be internal, and **our** parser would give us the exact position.
2. Adjust the error message provided by  `regex`:  extract the location information and what character is (un)expected. I don't know how robust this would be, you need to investigate the error messages. The best case scenario would be if `regex` package can provide this in a more structured way (e.g. dictionary).

Reply:
Even though the `regex` package offers structured exception informations I've decided to extend our grammar to "replicate" `regex` patterns (excluding some extensive niche options) as it seemed safer and more reliable. 
Implemented in ddd26fe.